widgets/secretProviders/src/RemoveSecretProviderButton.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import { tableRefreshEvent, translateSecretProviders } from './SecretProvidersTable.consts';
+import { RequestStatus } from './types';
+import type { SecretProvidersWidget } from './widget.types';
+
+const { Icon, Confirm: DeleteModal } = Stage.Basic;
+
+const { useBoolean } = Stage.Hooks;
+
+interface RemoveSecretProviderButtonProps {
+    secretProvider: SecretProvidersWidget.DataItem;
+    toolbox: Stage.Types.Toolbox;
+}
+
+const RemoveSecretProviderButton = ({ secretProvider, toolbox }: RemoveSecretProviderButtonProps) => {
+    const [isModalVisible, showModal, hideModal] = useBoolean();
+    const [deletingStatus, setDeletingStatus] = useState<RequestStatus>(RequestStatus.INITIAL);
+    const deleteModalContent = translateSecretProviders('deleteModal.content', {
+        secretProviderId: secretProvider.id
+    });
+
+    const removeSecretProvider = () => {
+        setDeletingStatus(RequestStatus.SUBMITTING);
+
+        toolbox
+            .getManager()
+            .doDelete(`/secrets-providers/${secretProvider.id}`)
+            .then(() => {
+                toolbox.getEventBus().trigger(tableRefreshEvent);
+            })
+            .catch(() => {
+                setDeletingStatus(RequestStatus.ERROR);
+            });
+    };
+
+    useEffect(() => {
+        if (deletingStatus === RequestStatus.SUBMITTED) {
+            hideModal();
+        }
+    }, [deletingStatus]);
```

Comment:
If we are using `deletingStatus` just to show/hide modal, there is not need for actually using it, see the example below 😉 

```suggestion
    const deleteModalContent = translateSecretProviders('deleteModal.content', {
        secretProviderId: secretProvider.id
    });

    const removeSecretProvider = () => {
        setDeletingStatus(RequestStatus.SUBMITTING);

        toolbox
            .getManager()
            .doDelete(`/secrets-providers/${secretProvider.id}`)
            .then(() => {
                hideModal();
                toolbox.getEventBus().trigger(tableRefreshEvent);
            })
    };
```

One would say that we also are removing error handling, but till now we were setting `RequestStatus.ERROR` flag, once an error would appear, but we weren't handling it anyhow 😁 
So the provided example has the same amount of anti-faulty mechanisms as the proposed solution 😉

Reply:
If the modification would seem okay for you, we should also remove `./types.ts` file, as its content wouldn't be useful anymore 💪

Reply:
implemented adjustments: https://github.com/cloudify-cosmo/cloudify-stage/pull/2315/commits/dc55e4bf73feaf22389713b03d31ac4fd97d5ed3 :)